app/services/bulk_affiliate_styles_uploader.rb

```diff
@@ -0,0 +1,205 @@
+# frozen_string_literal: true
+
+class BulkAffiliateStylesUploader
+  MAXIMUM_FILE_SIZE = 4.megabytes
+  VALID_CONTENT_TYPES = %w[text/csv].freeze
+  URL_ALREADY_TAKEN_MESSAGE = 'Validation failed: Url has already been taken'
+
+  attr_reader :results
+
+  class Error < StandardError
+  end
+
+  class Results
```

Comment:
Would we be able to extract this rather than have a nested class that is this big?